com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs

```diff
@@ -551,14 +551,14 @@ void UpgradeResourcesIfNeeded()
             if (HDRenderPipeline.defaultAsset.renderPipelineResources == null)
                 HDRenderPipeline.defaultAsset.renderPipelineResources
                     = UnityEditor.AssetDatabase.LoadAssetAtPath<RenderPipelineResources>(HDUtils.GetHDRenderPipelinePath() + "Runtime/RenderPipelineResources/HDRenderPipelineResources.asset");
-			ResourceReloader.ReloadAllNullIn(HDRenderPipeline.defaultAsset.renderPipelineResources, HDUtils.GetHDRenderPipelinePath());
+			ResourceReloader.TryReloadAllNullIn(HDRenderPipeline.defaultAsset.renderPipelineResources, HDUtils.GetHDRenderPipelinePath());
```

Comment:
This is not good. In HDRP, we assume all resources are loaded at construction of the render pipeline. There is no lazy init for resources afterward. So having not fully verified resources here will lead to issue afterward.

Reply:
Note: If this change required to have Linux working? Otherwise we can simply revert it. thanks

Reply:
I have done a Test with old code: "ReloadAllNullIn" and indeed it don't work on linux and produce a black screen...

https://github.com/Unity-Technologies/Graphics/pull/744

would be nice to know more about what we should do :)

Reply:
@RSlysz  I have looked at if we could find a way to not use lazy initialization for the resources however i could not get anything working on linux. somehow the asset loader can not load that resource at that time. if we dont catch those exceptions in the constructor it is caught at higher level and  the pipeline that is returned from the constructor becomes null. and there is no rendering happening.